compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -149,6 +151,25 @@ public BallerinaSemanticModel(BLangPackage bLangPackage, CompilerContext context
         return compiledSymbols;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<Location> references(String fileName, LinePosition position) {
```

Comment:
Use of references across the modules is a common case. Have we thought of how user would query such references across modules? How one could query a single symbol across multiple semantic models of the modules. Is it possible to allow query `references()` through a Symbol as input?

Reply:
I think @rasika 's concern is a valid requirement. There is a catch though, which is the performance. With the symbol being an input then we query the symbol API and then again query the references API.

Reply:
This is what we talked about having an integration layer between the project API and the semantic API for right? Can definitely add another API which accepts a symbol as the input. In fact, I tried that approach first, and then reverted to position. From a performance POV, I don't think it'll be an issue since we don't need to query the symbol API again.

Reply:
Since now we allow `Symbol` as input, i think rest of the query can be possible with the introduction of the integration layer.